packages/test/test-end-to-end-tests/src/test/gc/gcBlobInSummaries.spec.ts

```diff
@@ -0,0 +1,251 @@
+
+            // Summarize on a new summarizer client and validate that a GC blob handle is generated.
+            await submitSummaryAndValidateState(summarizerClient3, isBlobHandle);
+            const snapshot1 = await summarizerClient1.containerRuntime.storage.getSnapshotTree();
+            const snapshot2 = await summarizerClient2.containerRuntime.storage.getSnapshotTree();
+            const snapshot3 = await summarizerClient3.containerRuntime.storage.getSnapshotTree();
+            assert.deepEqual(snapshot1, snapshot2, "Snapshots between containers should be the same!");
+            assert.deepEqual(snapshot2, snapshot3, "Snapshots between containers should be the regardless of handle!");
+        });
+
+        it("New gc blobs are submitted when handles are added and deleted", async () => {
+            // Make a change in dataStoreA.
+            dataStoreA._root.set("key", "value");
+
+            // A gc blob handle should be submitted as there are no gc changes
+            await submitSummaryAndValidateState(summarizerClient1, isBlobHandle);
+
+            // A new gc blob should be submitted as there is a deleted gc reference
+            dataStoreA._root.delete("dataStoreC");
+
+            // Summarize and validate that all data store entries are trees since a datastore reference has changed.
+            await submitSummaryAndValidateState(summarizerClient1, isBlob);
+
+            // A gc blob handle should be submitted as there are no gc changes
+            await submitSummaryAndValidateState(summarizerClient1, isBlobHandle);
+
+            // Add a handle reference to dataStore C
+            dataStoreA._root.set("dataStoreC", dataStoreC.handle);
+            // A new gc blob should be submitted as there is a new gc reference
+            await submitSummaryAndValidateState(summarizerClient1, isBlob);
+        });
+
+        it("GC blob handle written when summary fails", async () => {
+            // Make a change in dataStoreA.
+            dataStoreA._root.set("key", "value");
+
+            // A gc blob handle should be submitted as there are no gc changes
+            await submitSummaryAndValidateState(summarizerClient1, isBlobHandle);
+
+            await submitFailingSummary(provider, summarizerClient1, logger, SubmitSummaryStage.Generate);
+
+            // GC blob handle expected
+            await submitSummaryAndValidateState(summarizerClient1, isBlobHandle);
+        });
+
+        // TODO - fix this failing test
+        it("GC blob written when summary fails", async () => {
+            // Make a reference change by deleting a handle
+            dataStoreA._root.delete("dataStoreB");
+
+            await provider.ensureSynchronized();
+
+            await submitFailingSummary(provider, summarizerClient1, logger, SubmitSummaryStage.Generate);
+
+            // GC blob expected as the summary had changed
+            await submitSummaryAndValidateState(summarizerClient1, isBlob);
```

Comment:
This fails as a blob handle is submitted instead of a blob.
High level steps
- We submit and ack a summary successfully.
- A new GC blob is generated as this is the first summary
- A datastore is unreferenced meaning a new GC blob should be submitted
- We submit a summary that is cancelled on the generate step but gc is run.
- We submit and ack a summary successfully.
- The summary should be the new GC blob.